lib/ruby_lsp/test_reporter.rb

```diff
@@ -67,6 +67,80 @@ def append_output(message:)
         send_message("append_output", params)
       end
 
+      # Gather the results returned by Coverage.result and format like the VS Code test explorer expects
+      #
+      # Coverage result format:
+      #
+      # Lines are reported in order as an array where each number is the number of times it was executed. For example,
+      # the following says that line 0 was executed 1 time and line 1 executed 3 times: [1, 3].
+      # Nil values represent lines for which coverage is not available, like empty lines, comments or keywords like
+      # `else`
+      #
+      # Branches are a hash containing the name of the branch and the location where it is found in tuples with the
+      # following elements: [NAME, ID, START_LINE, START_COLUMN, END_LINE, END_COLUMN] as the keys and the value is the
+      # number of times it was executed
+      #
+      # Methods are a similar hash [ClassName, :method_name, START_LINE, START_COLUMN, END_LINE, END_COLUMN] => NUMBER
+      # OF EXECUTIONS
+      #
+      # Example:
+      # {
+      #   "file_path" => {
+      #     "lines" => [1, 2, 3, nil],
+      #     "branches" => {
+      #       ["&.", 0, 6, 21, 6, 65] => { [:then, 1, 6, 21, 6, 65] => 0, [:else, 5, 7, 0, 7, 87] => 1 }
+      #     },
+      #     "methods" => {
+      #       ["Foo", :bar, 6, 21, 6, 65] => 0
+      #     }
+      # }
+      #: () -> Hash[String, untyped]
+      def gather_coverage_results
+        # Ignore coverage results inside dependencies
+        result = Coverage.result.reject do |file_path, _coverage_info|
+          file_path.start_with?(Bundler.bundle_path.to_s) ||
+            file_path.start_with?(File.dirname(RbConfig::CONFIG["rubylibdir"])) ||
```

Comment:
Is it worth hoisting this `File.dirname(RbConfig::CONFIG["rubylibdir"])` rather than recalculating for each file?